src/rdb.c

```diff
@@ -2811,56 +2796,68 @@ void rdbLoadProgressCallback(rio *r, const void *buf, size_t len) {
  *
  * The lib_ctx argument is also optional. If NULL is given, only verify rdb
  * structure with out performing the actual functions loading. */
-int rdbFunctionLoad(rio *rdb, int ver, functionsLibCtx* lib_ctx, int rdbflags, sds *err) {
+int rdbFunctionLoad(rio *rdb, int ver, functionsLibCtx* lib_ctx, int type, int rdbflags, sds *err) {
     UNUSED(ver);
+    sds error = NULL;
+    sds final_payload = NULL;
     sds name = NULL;
     sds engine_name = NULL;
     sds desc = NULL;
     sds blob = NULL;
     uint64_t has_desc;
```

Comment:
maybe we should move these 5 variables and their cleanup to inside the RDB_OPCODE_FUNCTION block?
if we do that, we should then use a separate goto label for errors inside this block, which will jump to the outer error label.

Reply:
Fixed